docs/release-notes/self-hosted/self-hosted-v12.0.0.md

```diff
@@ -0,0 +1,102 @@
+---
+rss_title: Codacy release notes RSS feed
+rss_href: /feed_rss_created.xml
+description: Release notes for Codacy Self-hosted v12.0.0.
+codacy_tools_version_old: https://github.com/codacy/codacy-tools-sh/releases/tag/sh-1.1.6
+codacy_tools_version_new: https://github.com/codacy/codacy-tools-sh/releases/tag/sh-1.2.2
+---
+
+# Self-hosted v12.0.0
+
+These release notes are for [Codacy Self-hosted v12.0.0](https://github.com/codacy/chart/releases/tag/12.0.0){: target="_blank"}, released on June 21, 2023.<!-- TODO Update release date -->
+
+📢 [Visit the Codacy roadmap](https://roadmap.codacy.com) and <span class="skip-vale">let us know</span> your feedback on both new and planned product updates!
+
+## Upgrading Codacy Self-hosted
+
+Follow the steps below to upgrade to Codacy Self-hosted v12.0.0:
+
+1.  Check the [release notes for all Codacy Self-hosted versions](../index.md#self-hosted) **between your current version and the most recent version** for breaking changes and follow the instructions provided <span class="skip-vale">carefully</span>.
+
+1.  Follow the instructions to [upgrade your Codacy Self-hosted instance](https://docs.codacy.com/v12.0/chart/maintenance/upgrade/).
+
+1.  Update your Codacy command-line tools to the following versions:<!--TODO Update CLI tool versions-->
+
+    -   [Codacy Analysis CLI MAJOR.MINOR.PATCH](https://github.com/codacy/codacy-analysis-cli/releases/tag/MAJOR.MINOR.PATCH)
+    -   [Codacy Coverage Reporter MAJOR.MINOR.PATCH](https://github.com/codacy/codacy-coverage-reporter/releases/tag/MAJOR.MINOR.PATCH)
+
+## Breaking changes
+
+-   If you're updating Kubernetes to v1.25, upgrade to Codacy Self-hosted v12.0.0 before doing so, as some Kubernetes endpoints have been deprecated and may lead to service outage if used with earlier versions of Codacy Self-hosted.
+
+-   If using the monitoring functionality [Grafana + Prometheus + Loki](https://docs.codacy.com/chart/configuration/monitoring/#setting-up-monitoring-using-grafana-prometheus-and-loki), ensure Pod Security Policies are disabled for Loki and Promtail by following the relevant update instructions ([Loki](https://docs.codacy.com/chart/configuration/monitoring/#2-installing-loki), [Promtail](https://docs.codacy.com/chart/configuration/monitoring/#3-installing-promtail)) after upgrading to Codacy Self-hosted v12.0.0 and before updating Kubernetes to v1.25.
+
+-   If you have set `scheduler.enable = "true"` in `values.yaml`, before migrating to Codacy Self-hosted v12.0.0:
+
+    -   Remove scheduler settings from `values.yaml`
+    -   In `worker-manager`, ConfigMap set CONFIG_FORCE_codacy_kubernetes_scheduler_name: default-scheduler 
+    -   Restart `worker-manager` deployment
+    -   Remove any pending worker Pods
```

Comment:
These instructions read like a draft. Is there any chance we can make them more explicit for customers?

For example:

- Remove scheduler settings from `values.yaml` :arrow_right: perhaps another way of saying this could be to remove all `scheduler.*` entries from the file?
- In `worker-manager`, ConfigMap set CONFIG_FORCE_codacy_kubernetes_scheduler_name: default-scheduler :arrow_right: what does this actually mean? What is `worker-manager`? Is it a pod? The "ConfigMap set [...]" looks like it could be a command... If it is, how do users run it?
- Restart `worker-manager` deployment :arrow_right: Can we provide a command for doing this?
- Remove any pending worker Pods :arrow_right: How do users check if there are "pending worker pods"? Which pods are those? Again, is there a command we can provide to remove those pods?

@tfrgoncalves Perhaps you can help clarify some of these questions?

Reply:
1. Make sense, indeed. 
2. [Here](https://github.com/codacy/docs/pull/1767#discussion_r1239966685), I wrote some other options for us to write it programatically, we just need to choose one. In this case the ConfigMap is called `worker-manager`, and is used by a `worker-manager` deployment.
3. `kubectl rollout restart deployment/worker-manager -n codacy`
4. `kubectl get pods -l app=worker --field-selector=status.phase=Pending -n codacy` if result is empty no pods are pending.

Let me know if any of these need further clarification.

Reply:
@tfrgoncalves I updated the [breaking changes instructions](https://feat-release-notes-self-hosted-12-0--docs-codacy.netlify.app/release-notes/self-hosted/self-hosted-v12.0.0/#breaking-changes).

* Please take a look and let us know if everything is correct.
* We'll also need the final command to delete pending worker pods. I understand that this is a destructive action, and added a warning before it.

Thanks!